server/src/main/java/org/candlepin/resource/EntitlementResource.java

```diff
@@ -152,18 +125,24 @@ public EntitlementDTO hasEntitlement(@PathParam("consumer_uuid") String consumer
                 consumerUuid, productId));
     }
 
-    @ApiOperation(notes = "Retrieves list of Entitlements", value = "listAllForConsumer")
-    @ApiResponses({ @ApiResponse(code = 400, message = "") })
-    @GET
-    @Produces(MediaType.APPLICATION_JSON)
+    @Override
     public List<EntitlementDTO> listAllForConsumer(
-        @QueryParam("consumer") String consumerUuid,
-        @QueryParam("matches") String matches,
-        @QueryParam("attribute") List<KeyValueParameter> attrFilters,
-        @Context PageRequest pageRequest) {
+        String consumerUuid,
+        String matches,
+        List<KeyValueParamDTO> attrFilters,
+        Integer page,
+        Integer perPage,
+        String order,
+        String orderBy) {
```

Comment:
One problem I see with this option is that paging will be hidden from Swagger documentation. We could document it in the description but I am not sure how effective that will be..

Reply:
That's a good point. Even with our current state the PageRequest is hidden from swagger documentation, so if it could be included it would be a good thing. @Ceiu do you see anything wrong with this approach?

If we are going to do this though, we should probably add some paging spec tests in entitlement_resource_spec.rb to make sure functionality has not changed (it has occurred to me we don't have any paging tests at all, so this would be a good moment to add some)

Reply:
On one hand, I'm hesitant to make more sweeping changes because of limitations in OpenAPI. As it is today, paging "just works" so long as the endpoint passes the paging info through (which, admittedly, requires varying degrees of extra complexity). This is very similar to include/exclude, which are also optional parameters available globally that are probably going undocumented by things like Swagger.

On the other hand, because the endpoints need to add extra logic to support paging, I see value in having explicit documentation on the endpoints themselves indicating as much, rather than having it as a global "this exists and it may or may not work, good luck" note somewhere.

Overall, I feel that if we're going to address the plate of spaghetti that is paging, then we should do that. But if we're not in a place where we can fully deal with it, then we should probably keep pulling it from the context so our defaults and constraints are consistent with every other endpoint that uses paging today.

As far as testing goes, some of the endpoints have unit tests for the paging bits, so those likely aren't as in-need of spec testing as the ones that aren't tested at all.

Reply:
Alright, I would prefer not to do sweeping changes, since this epic is about simply moving our API to openapi. If we want to overhaul the paging mechanism (and in the process make it better-documented by including it in the openapi spec file), we can add it as a future card.

So for now I'd say just remove it as an argument and add `PageRequest pageRequest = ResteasyProviderFactory.getContextData(PageRequest.class);`. For making sure it is documented, what do you think about adding something to the endpoint's description, along the lines of `This endpoint supports paging with query parameters. Please see here for details: https://www.candlepinproject.org/docs/candlepin/pagination.html#paginating-results-from-candlepin`. And maybe we can do the same thing for all other endpoints that support paging?

@Januson one more thing: I noticed that PageRequestDTO is not used anywhere anyway (paging arguments are specified separately), so it should have been removed either way.